src/formatDate.js

```diff
@@ -50,7 +50,43 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[fromFormat.length - 1]);
+  let year, month, day;
+
+  for (let i = 0; i < dateArray.length; i++) {
```

Comment:
Instead of using a loop to assign date components, you can use 'findIndex' method to find the index of the component in the 'fromFormat' array and use that index to get the value from 'dateArray'. This will simplify the code and improve readability.

Reply:
Okay , thank you